### ISSUE TITLE
New check for SQL logins with blank or Login as password

### ISSUE THREAD

**Issue by PsPsam** [sqlcollaborative/dbachecks#413]
## Issue

This is a new check/question
Idea taken from SQL Assessment in azure 
  
<-- Simply answer the question - What would you like dbachecks to do that it doesn't? -->   
   
## New Check
   I would like a test for logins with no password and password same as username

### What would you like to check?
The test should return the count of the result. So if all is good 0 would be returned.

---Detect SQL Logins with weak passwords

SELECT count(name)
FROM master.sys.sql_logins
WHERE PWDCOMPARE(name,password_hash)=1
GO

and 
---Detect SQL Logins with blank passwords

SELECT count(name)
FROM master.sys.sql_logins
WHERE PWDCOMPARE('',password_hash)=1
GO
  

### What should be configurable for the results of the check ?
If test should run or not     
<-- A value, a set of properties (like FULL, SIMPLE, BULKLOGGED) -->  
<-- Should there be any default values for the configs? -->   
    
### What should be able to be excluded from being tested ?
Nothing    
<-- Could be databases, Could be Availability Groups etc -->  
<-- This would not be instances -->  
   
## Additional information    
I took a quick look at dbatools but saw no command that does this. So maybe it a command should be created there first for this?

**Comment by SQLDBAWithABeard** [comment]
I like this test, do you know if this T-SQL will work on all versions of SQL Server?

**Comment by PsPsam** [comment]
Its from sql 2008 and up 

https://docs.microsoft.com/en-us/sql/t-sql/functions/pwdcompare-transact-sql


I changed the t-sql to better align with pester test, but you can get more information. So im not sure if this should be a dbatool test function that we than can test agains in dbachecks

The origial t-sql is 
```
---Detect SQL Logins with weak passwords 
SELECT SERVERPROPERTY('machinename') 
AS 'Server Name', 
ISNULL(SERVERPROPERTY ('instancename'), 
SERVERPROPERTY ('machinename')) AS 'Instance Name', 
name AS 'Login With Password Equal to Login Name' 
FROM master.sys.sql_logins 
WHERE PWDCOMPARE(name,password_hash)=1 
ORDER BY name 
GO
``` 
From azure

**Comment by SQLDBAWithABeard** [comment]
I agree in principle that this can be added, we would need some logic to skip the test if the version is below 2008

**Comment by michalporeba** [comment]
I was just wondering what is the value of including this function in dbatools rather than implementing it in dbachecks? especially the proposed implementation simply executes an t-sql code? Does it introduce unnecessary coupling? Would the `Test-DbaLoginPassword` ever get executed from dbatools directly?

**Comment by SQLDBAWithABeard** [comment]
Yes it undoubtedly would and doing so will bring more capabilities than just checking
